client/src/test/java/de/hellfish/ngrok/client/NgrokClientTest.java

```diff
@@ -20,59 +25,74 @@
  * <p>
  * Simultaneously the test starts the Ngrok client, which connects to
  * this server and awaits a link message.
- * After server shutdown, Ngrok client catches Socket exception and interrupt its thread
+ * After server shutdown, Ngrok client catches Socket exception and interrupts its thread
  */
 @Slf4j
 public class NgrokClientTest {
 
-    private ServerSocket testServerSocket;
     private NgrokClient client;
-    private final int testPort = 8082;
+    private final int serverPort = 8085;
     private final String testLink = "http://w56h.myngrok.com";
+    private final int servicePort = 2224;
+    private final String testHeaderKey = "content-type";
+    private final String testHeaderValue = "text/plain";
+    private final String testMethod = "POST";
+    private final String testBody = "Test body";
 
     @BeforeEach
     public void setup() throws IOException, InterruptedException {
+        // Test service
+        new Thread(() -> {
+            try (ServerSocket serviceServerSocket = new ServerSocket(servicePort);
+                 Socket serviceSocket = serviceServerSocket.accept();
+                 BufferedReader in = new BufferedReader(new InputStreamReader(serviceSocket.getInputStream()));
+                 PrintWriter out = new PrintWriter(serviceSocket.getOutputStream(), true)
+            ) {
+                in.readLine();
+                out.println("HTTP/1.1 200 OK\r\n"
```

Comment:
Why do you need to construct HTTP requests manually? You can probably use HTTP client class.

Reply:
Yes, I also thought about it, but decided to leave it. In order to have different ways in one project - for study purposes.